MagicModules/TemplateMagicModulesInput.ts

```diff
@@ -393,7 +462,11 @@ function appendOption(output: string[], option: ModuleOption, isGo: boolean, isP
             dataType = "!ruby/object:Api::Azure::SDKTypeDefinition::ISO8601DateTimeObject";
             break;
         case "number":
-            dataType = "!ruby/object:Api::Azure::SDKTypeDefinition::IntegerObject";
+            if (option.IsList) {
+                dataType = "!ruby/object:Api::Azure::SDKTypeDefinition::Integer32ArrayObject";
```

Comment:
and this is mysterious, if it's already in the list, why we need this new type here?